tiatoolbox/models/dataset/interactive_segmentation.py

```diff
@@ -0,0 +1,156 @@
+
+        Args:
+            img_path (:obj:`str` or :obj:`pathlib.Path`): Path to a standard image,
+                a whole-slide image or a large tile to read.
+            points (ndarray, pd.DataFrame, str, pathlib.Path): Points ('clicks') for the image. 
+            label: A label. Default is `None`.
+            mode (str): Type of the image to process. Choose from either `patch`, `tile`
+                or `wsi`.
+            resolution (int or float or tuple of float): resolution at
+                which to read the image, default = 0. Either a single
+                number or a sequence of two numbers for x and y are
+                valid. This value is in terms of the corresponding
+                units. For example: resolution=0.5 and units="mpp" will
+                read the slide at 0.5 microns per-pixel, and
+                resolution=3, units="level" will read at level at
+                pyramid level / resolution layer 3.
+            units (str): the units of resolution, default = "level".
+                Supported units are: microns per pixel (mpp), objective
+                power (power), pyramid / resolution level (level),
+                Only pyramid / resolution levels (level) embedded in
+                the whole slide image are supported.
+            patch_size: Size of the patch to extract, default = (128, 128)
+
+        Examples:
+            >>> # create a dataset to extract small patches around each point on a patch image
+            >>> ds = InteractiveSegmentorDataset(
+            ...     img_path = 'example_image.png',
+            ...     points = 'example_points.csv',
+            ...     mode = 'patch',         
+            ... )
+
+        """
+        # Not using IOConfig at the moment
+
+        super().__init__()
+
+        if not os.path.isfile(img_path):
+            raise ValueError("`img_path` must be a valid file path.")
+        if mode not in ["patch", "wsi", "tile"]:
+            raise ValueError(f"`{mode}` is not supported.")
+
+        self.img_path = img_path
+        self.label = label
+        self.mode = mode
+        self.patch_size = patch_size    
+        self.resolution = resolution    
+        self.units = units             
+
+        # Read the points('clicks') into a panda df
+        self.locations = read_locations(points)
+
+        self.patch_extractor = get_patch_extractor("point",  
+            input_img = self.img_path, locations_list = points, patch_size=self.patch_size,
+            resolution = self.resolution, units = self.units)
+
+    def __getitem__(self, idx):
+        patch = self.patch_extractor.__getitem__(idx)
+
+        boundingBox = self.get_boundingBox(idx)
+
+        # we know nucPoint is at the centre of the patch:
+        nucPoint = np.zeros((1, self.patch_size[1], self.patch_size[1]), dtype=np.uint8)
+        nucPoint[0,int((self.patch_size[1]-1)/2),int((self.patch_size[1]-1)/2)] = 1
```

Comment:
[0,int((self.patch_size[**0**]-1)/2),int((self.patch_size[1]-1)/2)]

this first patch_size should be self.patch_size[**0**] because size of first and second dimensions might not be the same.